syntax/std_seq_prefix_test.go

```diff
@@ -63,3 +63,65 @@ func TestBytesPrefix(t *testing.T) {
 	AssertCodesEvalToSameValue(t, `false`, `//seq.has_prefix(<<'o'>>, <<>>)    `)
 	AssertCodesEvalToSameValue(t, `true `, `//seq.has_prefix(<<>>, <<'hello'>>)`)
 }
+
+func TestStrTrimPrefix(t *testing.T) {
+	t.Parallel()
+	AssertCodesEvalToSameValue(t, `""`, `//seq.trim_prefix("ABC","ABC")`)
+
+	AssertCodesEvalToSameValue(t, `"BCDE"`, `//seq.trim_prefix("A","ABCDE")`)
+	AssertCodesEvalToSameValue(t, `"CDE"`, `//seq.trim_prefix("AB","ABCDE")`)
+	AssertCodesEvalToSameValue(t, `"ABCDE"`, `//seq.trim_prefix("BCD","ABCDE")`)
+	AssertCodesEvalToSameValue(t, `"ABCDE"`, `//seq.trim_prefix("CD","ABCDE")`)
+	AssertCodesEvalToSameValue(t, `""`, `//seq.trim_prefix("CD","")`)
+	AssertCodesEvalToSameValue(t, `"ABCD"`, `//seq.trim_prefix("","ABCD")`)
+
+	AssertCodesEvalToSameValue(t, `""`, `//seq.trim_prefix("","")`)
+	AssertCodesEvalToSameValue(t, `""`, `//seq.trim_prefix("A","")`)
+	AssertCodesEvalToSameValue(t, `"A"`, `//seq.trim_prefix("","A")`)
+
+	AssertCodeErrors(t, `//seq.trim_prefix(1,"ABC")`, "")
+}
+
+func TestArrayTrimPrefix(t *testing.T) { //nolint:dupl
+	t.Parallel()
+	AssertCodesEvalToSameValue(t, `[]`, `//seq.trim_prefix(['A'],['A'])`)
+	AssertCodesEvalToSameValue(t, `[]`, `//seq.trim_prefix(['A','B'],['A','B'])`)
+
+	AssertCodesEvalToSameValue(t, `['B','C','D','E']`, `//seq.trim_prefix(['A']            ,['A','B','C','D','E'])`)
+	AssertCodesEvalToSameValue(t, `    ['C','D','E']`, `//seq.trim_prefix(['A','B']        ,['A','B','C','D','E'])`)
+	AssertCodesEvalToSameValue(t, `        ['D','E']`, `//seq.trim_prefix(['A','B','C']    ,['A','B','C','D','E'])`)
+	AssertCodesEvalToSameValue(t, `            ['E']`, `//seq.trim_prefix(['A','B','C','D'],['A','B','C','D','E'])`)
+	AssertCodesEvalToSameValue(t, `['A','B','C','D','E']`, `//seq.trim_prefix(['B'],['A','B','C','D','E'])`)
+	AssertCodesEvalToSameValue(t, `['A','B','C','D','E']`, `//seq.trim_prefix(['B','C'],['A','B','C','D','E'])`)
+	AssertCodesEvalToSameValue(t,
+		`['A','B','C','D','E']`,
+		`//seq.trim_prefix(['A','B','C','D','E','F'],['A','B','C','D','E'])`)
+	AssertCodesEvalToSameValue(t,
+		`['A','B','C','D','E']`,
+		`//seq.trim_prefix(['A','B','C','D','E','F'],['A','B','C','D','E'])`)
+
+	AssertCodesEvalToSameValue(t, `[3]`, `//seq.trim_prefix([1, 2],[1, 2, 3])`)
+	AssertCodesEvalToSameValue(t, `[[3]]`, `//seq.trim_prefix([[1, 2]],[[1, 2], [3]])`)
+
+	AssertCodesEvalToSameValue(t, `[]`, `//seq.trim_prefix([], [])`)
+	AssertCodesEvalToSameValue(t, `[]`, `//seq.trim_prefix(['A','B','C','D','E','F'],[])`)
+	AssertCodesEvalToSameValue(t, `['A','B','C','D','E']`, `//seq.trim_prefix([],['A','B','C','D','E'])`)
+
+	AssertCodeErrors(t, `//seq.trim_prefix(1,[1,2,3])`, "")
+	AssertCodeErrors(t, `//seq.trim_prefix('A',['A','B','C'])`, "")
```

Comment:
ditto

Reply:
sorry, meant it's just like this https://github.com/arr-ai/arrai/pull/358#discussion_r433047905